internal/models/argo.go

```diff
@@ -168,6 +187,88 @@ func (app *Application) ListUnhealthyResources() []string {
 	return list
 }
 
+// IsManagedByWatcher checks if the application is managed by the watcher.
+// It checks if the application's metadata contains the "argo-watcher/managed" annotation with the value "true".
+func (app *Application) IsManagedByWatcher() bool {
+	if app.Metadata.Annotations == nil {
+		return false
+	}
+	return app.Metadata.Annotations[managedAnnotation] == "true"
+}
+
+func (app *Application) processAppAnnotations(annotations map[string]string, task *Task) *updater.ArgoOverrideFile {
+	overrideFileContent := updater.ArgoOverrideFile{}
+	managedImages := extractManagedImages(annotations)
+
+	if len(managedImages) == 0 {
+		log.Error().Msgf("%s annotation not found, skipping image update", managedImagesAnnotation)
+		return nil
+	}
+
+	for _, image := range task.Images {
+		for appAlias, appImage := range managedImages {
+			if image.Image == appImage {
+				if tagPath, exists := annotations[fmt.Sprintf(managedImageTagPattern, appAlias)]; exists {
+					overrideFileContent.Helm.Parameters = append(overrideFileContent.Helm.Parameters, updater.ArgoParameterOverride{
+						Name:        tagPath,
+						Value:       image.Tag,
+						ForceString: true,
+					})
+				} else {
+					log.Error().Msgf("%s annotation not found, skipping image %s update", fmt.Sprintf(managedImageTagPattern, appAlias), image.Image)
+				}
+			}
+		}
+	}
+
+	return &overrideFileContent
+}
+
+func (app *Application) UpdateGitImageTag(task *Task) {
```

Comment:
I think this method should return an error. Then the error should propagate to the very top and fail the deployment / task. Otherwise - the client would be waiting for us to do the deployment, but we silently fail and then later in the code - wait for the updated version. 

It also doesn't make sense because we know that setting the new version failed, so there's no point in waiting for it to change in argo cd.